Translate createPortal to spanish (#619)
reactjs/es.react.dev: beta/src/content/reference/react-dom/createPortal.md

---

## Uso {/*usage*/}

### Renderizar a una parte diferente del DOM {/*rendering-to-a-different-part-of-the-dom*/}

*Portals* deja a tus componentes renderizar a algunos de sus hijos en un lugar diferente en el DOM. Esto permite que una parte de su componente "escape" de cualquier contenedor en el que pueda estar. Por ejemplo, un componente puede mostrar un modal de diálogo o información sobre herramientas que aparece encima y fuera del resto de la página.

Para crear un portal, renderizar el resultado de `createPortal` con <CodeStep step={1}>algo de JSX</CodeStep> y el <CodeStep step={2}>nodo del DOM donde debería ir</CodeStep>:

```js [[1, 8, "<p>This child is placed in the document body.</p>"], [2, 9, "document.body"]]
import { createPortal } from 'react-dom';

function MyComponent() {
  return (
    <div style={{ border: '2px solid black' }}>
      <p>Este hijo se coloca en el div padre.</p>
      {createPortal(
        <p>Este hijo es colocado en el cuerpo del documento.</p>,
        document.body
      )}
    </div>
  );
}
```

> React pondrá los nodos del DOM en <CodeStep step={1}>el JSX que pasaste</CodeStep> dentro del <CodeStep step={2}>nodo del DOM que proporcionó</CodeStep>. Sin un portal, el segundo `<p>` sería colocado dentro del padre `<div>`, pero el portal lo "teleported" al [`document.body`:](https://developer.mozilla.org/en-US/docs/Web/API/Document/body)

Reviewer: ```suggestion
React pondrá los nodos del DOM en <CodeStep step={1}>el JSX que pasaste</CodeStep> dentro del <CodeStep step={2}>nodo del DOM que proporcionó</CodeStep>. Sin un portal, el segundo `<p>` sería colocado dentro del padre `<div>`, pero el portal lo "teletransportará" al [`document.body`:](https://developer.mozilla.org/en-US/docs/Web/API/Document/body)
```

</Sandpack>

> Observe cómo el segundo párrafo aparece visualmente fuera del padre `<div>` sin el borde. Si inspecciona la estructura del DOM con las herramientas de desarrollador, puedes confirmar que el segundo `<p>` se colocó directamente en el `<body>`:

Reviewer: ```suggestion
Observa cómo el segundo párrafo aparece visualmente fuera del padre `<div>` sin el borde. Si inspeccionas la estructura del DOM con las herramientas de desarrollador, puedes confirmar que el segundo `<p>` se colocó directamente en el `<body>`:
```

```html {4-6,9}
<body>
  <div id="root">
    ...
      <div style="border: 2px solid black">
        <p>Este hijo se coloca dentro del div padre.</p>
      </div>
    ...
  </div>
  <p>Este hijo es colocado en el cuerpo del documento.</p>
</body>
```

Un portal solo cambia la ubicación física del nodo del DOM. En todos los demás sentidos, el JSX renderizado en un portal actúa como un nodo hijo del componente React que lo renderiza. Por ejemplo, el hijo puede acceder al contexto proporcionado por el padre, y los eventos siguen apareciendo de los hijos a los padres de acuerdo con el árbol React.

---

### Renderizado de un diálogo de un modal con un portal {/*rendering-a-modal-dialog-with-a-portal*/}

> Puede usar un portal para crear un diálogo con un modal que flote sobre el resto de la página, incluso si el componente que convoca el diálogo está dentro de un contenedor con `overflow: hidden` u otros estilos que interfieren con el diálogo.

Reviewer: ```suggestion
Puedes usar un portal para crear un diálogo con un modal que flote sobre el resto de la página, incluso si el componente que convoca el diálogo está dentro de un contenedor con `overflow: hidden` u otros estilos que interfieren con el diálogo.
```

<Pitfall>

> Es importante asegurarse de que su aplicación sea accesible cuando use portales. Por ejemplo, es posible que deba administrar el enfoque del teclado para que el usuario pueda mover el enfoque dentro y fuera del portal de forma natural.

Reviewer: ```suggestion
Es importante asegurarse de que tu aplicación sea accesible cuando use portales. Por ejemplo, es posible que deba administrar el enfoque del teclado para que el usuario pueda mover el enfoque dentro y fuera del portal de forma natural.
```

> Siga el [WAI-ARIA Modal Authoring Practices](https://www.w3.org/WAI/ARIA/apg/#dialog_modal) al crear modales. Si utiliza un paquete comunitario, asegúrese de que sea accesible y siga estas pautas.

Reviewer: ```suggestion
Sigue el [WAI-ARIA Modal Authoring Practices](https://www.w3.org/WAI/ARIA/apg/#dialog_modal) al crear modales. Si utilizas un paquete comunitario, asegúrate de que sea accesible y siga estas pautas.
```

</Pitfall>

---

### Renderizado de componentes de React en marcado de servidor que no es de React {/*rendering-react-components-into-non-react-server-markup*/}

> Los portales pueden ser útiles si su raíz de React es solo parte de una página estática o renderizada por un servidor que no está construida con React. Por ejemplo, si su página está construida con un framework de servidor como Rails o PHP, puede crear áreas interactivas dentro de áreas estáticas como barras laterales. Comparado con tener [multiple separate React roots,](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) Los portales le permiten tratar la aplicación como un solo árbol React con estado compartido, aunque sus partes se representen en diferentes partes del DOM.

Reviewer: ```suggestion
Los portales pueden ser útiles si tu raíz de React es solo parte de una página estática o renderizada por un servidor que no está construida con React. Por ejemplo, si tu página está construida con un framework de servidor como Rails o PHP, puedes crear áreas interactivas dentro de áreas estáticas como barras laterales. Comparado con tener [multiple separate React roots,](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) Los portales te permiten tratar la aplicación como un solo árbol React con estado compartido, aunque sus partes se rendericen en diferentes partes del DOM.
```

---

### Renderizado de componentes de React en nodos del DOM que no son de React {/*rendering-react-components-into-non-react-dom-nodes*/}

> También puede usar un portal para administrar el contenido de un nodo del DOM que se administra fuera de React. Por ejemplo, suponga que se está integrando con un widget de un mapa que no es de React y desea mostrar el contenido de React dentro de una ventana emergente.

Reviewer: ```suggestion
También puedes usar un portal para administrar el contenido de un nodo del DOM que se administra fuera de React. Por ejemplo, supón que te estás integrando con un widget de un mapa que no es de React y desea mostrar el contenido de React dentro de una ventana emergente.
```

> Para ello, declara un `popupContainer` variable de estado para almacenar el nodo del DOM en el que se va a representar:

Reviewer: ```suggestion
Para ello, declara una variable de estado `popupContainer` para almacenar el nodo del DOM en el que se va a renderizar:
```
